### PR TITLE
feat:  Docker adds ca-certificate

### PR DESCRIPTION
**Pull request by soluchok**
Adds ca-certificate.

Issue: When we communicate between two services(in docker) there is a problem with invalid certs. This change makes the certs `trustable`.

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: test/bdd/fixtures/hub-router/docker-compose.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ services:
       - ${HUB_ROUTER_DIDCOMM_HTTP_PORT}:${HUB_ROUTER_DIDCOMM_HTTP_PORT}
       - ${HUB_ROUTER_DIDCOMM_WS_PORT}:${HUB_ROUTER_DIDCOMM_WS_PORT}
     entrypoint: ""
-    command: /bin/sh -c "hub-router start"
+    command: /bin/sh -c "cp /etc/tls/* /usr/local/share/ca-certificates/;update-ca-certificates; hub-router start"
```
Comment (llorllale) [inline test/bdd/fixtures/hub-router/docker-compose.yml:32]: We typically mount the CA certs into a volume and pass them as configuration options to the application. I cannot say with certainty why we're doing it that way but I like this method since it does not require extra code on our part.

Comment (llorllale) [reply test/bdd/fixtures/hub-router/docker-compose.yml:32]: The host OS' configuration is never touched because we always run the binaries inside docker containers.

**Review by llorllale — APPROVED** [review]

File: test/bdd/fixtures/hub-router/docker-compose.yml (hunk 1 above)
Comment (soluchok) [reply test/bdd/fixtures/hub-router/docker-compose.yml:32]: @llorllale when we communicate between two services(in docker) there is a problem with invalid certs. This change makes the certs `trustable`.
I've got this error in `edge-agent`. Routers are not able to communicate with each other due to ` invalid certs` error.

Comment (llorllale) [reply test/bdd/fixtures/hub-router/docker-compose.yml:32]: @soluchok that sounds about right. Sorry if my comment distracted from the motivation behind this PR. Maybe you should include that comment in this PR's description.

My comment above was just highlighting how we configure the CA certs differently, but I like this way better because it is effective and involves less work.

**Review by fqutishat — CHANGES_REQUESTED** [review]
you need to add tls ca like this https://github.com/trustbloc/edge-sandbox/blob/master/test/bdd/fixtures/demo/docker-compose-demo-applications.yml#L29

**Comment by soluchok** [comment]
> you need to add tls ca like this https://github.com/trustbloc/edge-sandbox/blob/master/test/bdd/fixtures/demo/docker-compose-demo-applications.yml#L29

@fqutishat It was already there https://github.com/trustbloc/hub-router/blob/79110fbc6f85abc7e59c07d21e098ceeefd60e10/test/bdd/fixtures/hub-router/docker-compose.yml#L18

**Comment by llorllale** [comment]
@fqutishat 

> you need to add tls ca like this https://github.com/trustbloc/edge-sandbox/blob/master/test/bdd/fixtures/demo/docker-compose-demo-applications.yml#L29

Please explain the reasoning behind that way of configuring the CA certs, because it requires more effort to implement than the solution presented here.

**Comment by rolsonquadras** [comment]
> @fqutishat
> 
> > you need to add tls ca like this https://github.com/trustbloc/edge-sandbox/blob/master/test/bdd/fixtures/demo/docker-compose-demo-applications.yml#L29
> 
> Please explain the reasoning behind that way of configuring the CA certs, because it requires more effort to implement than the solution presented here.

I believe it's to reduce the exposure of untrusted certs (process vs system/os).

**Comment by rolsonquadras** [comment]
> Adds ca-certificate.
> 
> Issue: When we communicate between two services(in docker) there is a problem with invalid certs. This change makes the certs `trustable`.
> 
> Signed-off-by: Andrii Soluk [isoluchok@gmail.com](mailto:isoluchok@gmail.com)

@soluchok Where does this fail (what's repo/bddtests) ? Currently, TLS is used with self-signed certs in hub-router bdd tests (TLS enabled [adapter](https://github.com/trustbloc/hub-router/blob/master/test/bdd/fixtures/integration/docker-compose.yml#L55) talks to TLS enabled [hub-router](https://github.com/trustbloc/hub-router/blob/master/test/bdd/fixtures/hub-router/docker-compose.yml#L20) through DIDComm from within docker containers) and also tested in edge-sandbox. Just wanted to know if I missed something.

**Comment by fqutishat** [comment]
@troyronda @sudeshrshetty can you post your opinion

**Comment by llorllale** [comment]
@rolsonquadras 

> > @fqutishat
> > > you need to add tls ca like this https://github.com/trustbloc/edge-sandbox/blob/master/test/bdd/fixtures/demo/docker-compose-demo-applications.yml#L29
> > 
> > 
> > Please explain the reasoning behind that way of configuring the CA certs, because it requires more effort to implement than the solution presented here.
> 
> I believe it's to reduce the exposure of untrusted certs (process vs system/os).

We still have the same risk profile with this setup. The container's operator still needs to configure the CA certs somehow, and they would keep on doing what they are doing today: mounting the volume. We're just removing the extra step of maintaining a config variable inside the container's environment and in our code.

**Comment by rolsonquadras** [comment]
@llorllale 
> @rolsonquadras
> 
> > > @fqutishat
> > > > you need to add tls ca like this https://github.com/trustbloc/edge-sandbox/blob/master/test/bdd/fixtures/demo/docker-compose-demo-applications.yml#L29
> > > 
> > > 
> > > Please explain the reasoning behind that way of configuring the CA certs, because it requires more effort to implement than the solution presented here.
> > 
> > 
> > I believe it's to reduce the exposure of untrusted certs (process vs system/os).
> 
> We still have the same risk profile with this setup. The container's operator still needs to configure the CA certs somehow, and they would keep on doing what they are doing today: mounting the volume. We're just removing the extra step of maintaining a config variable inside the container's environment and in our code.

If we inject trusted cert in the code, its just exposed to app (I believe that’s what happening). If we update system CA to add, we will be exposing the untrusted certs systemwide (someone can login to docker and curl). Usually access to docker bash is restricted. I don’t have strong opinion (both are ok to me).
